### PR TITLE
Added azure pipeline example for Kubeflow 

### PR DESCRIPTION
**Pull request by rem20806**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/examples/599)

### CODE REVIEW

**Comment by rem20806** [comment]
@sarahmaddox Thank you for your help! docs coming soon

**Comment by rem20806** [comment]
@sethjuarez is new to the Kubeflow repo but he says he has signed the CLA

**Comment by rem20806** [comment]
> @rem20806: The following test **failed**, say `/retest` to rerun them all:
> 
> Test name	Commit	Details	Rerun command
> kubeflow-examples-presubmit	[3b85d55](https://github.com/kubeflow/examples/commit/3b85d55d8c28c5c2747042e40675ec61413e1894)	[link](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubeflow_examples/599/kubeflow-examples-presubmit/1151615057201729536/)	`/test kubeflow-examples-presubmit`
> [Full PR test history](https://prow.k8s.io/pr-history?org=kubeflow&repo=examples&pr=599). [Your PR dashboard](https://gubernator.k8s.io/pr/rem20806). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubeflow/examples/issues?q=is:issue+is:open) when you hit one in your PR.
> 
> Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md). If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).

These tests were failed because the bot is trying to run this pipeline using gcloud commands. This pipeline is designed to show how azure ml functions can work on kubeflow pipelines. The docs for how to use this example are being written currently. Is there any way to bypass the automated testing? @karthikv2k @IronPan

**Comment by sarahmaddox** [comment]
Part of https://github.com/kubeflow/website/issues/880 - docs coming soon.

**Comment by rem20806** [comment]
Docs have been pushed kubeflow/website#956

**Comment by texasmichelle** [comment]
Thanks for creating this and the corresponding docs! This is a large amount of files - it looks like you're checking in a ksonnet app, which has been deprecated. Is there a reason kustomize isn't a good fit?

**Comment by texasmichelle** [comment]
As for the CLA failures, me@sethjuarez.com needs to sign the CLA and t-moahme@microsoft.com and maanavdalal@gmail.com need to provide consent.

**Comment by rem20806** [comment]
> As for the CLA failures, [me@sethjuarez.com](mailto:me@sethjuarez.com) needs to sign the CLA and [t-moahme@microsoft.com](mailto:t-moahme@microsoft.com) and [maanavdalal@gmail.com](mailto:maanavdalal@gmail.com) need to provide consent.

**Comment by rem20806** [comment]
@texasmichelle my apologies. There were some folders in the PR that were not being used. The ksapp is one of those. I will make the change asap

**Comment by MaanavD** [comment]
I consent.

**Comment by MaanavD** [comment]
@googlebot I consent.

**Comment by rem20806** [comment]
Also, @texasmichelle, let me know if there is anything I can do to help you through testing. If you run into any errors, we should probably try to fix them in the docs first.

File: pipelines/azurepipeline/LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (texasmichelle) [inline pipelines/azurepipeline/LICENSE]: This file already exists in the repo base dir - no need to include it here.

File: pipelines/azurepipeline/code/deploy/inference.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+echo "test the deployment with a burrito image"
```
Comment (texasmichelle) [inline pipelines/azurepipeline/code/deploy/inference.sh]: Add a `#!` statement so these can be run directly

File: pipelines/azurepipeline/code/preprocess/build.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+IMAGE=kubeflowregistry.azurecr.io/kubeflow/preprocess
```
Comment (texasmichelle) [inline pipelines/azurepipeline/code/preprocess/build.sh]: Add `#!` line, e.g. `#!/bin/bash`

File: pipelines/azurepipeline/code/register/build.sh [hunk 4]
```diff
@@ -0,0 +1,2 @@
+IMAGE=tacoregistry.azurecr.io/kubeflow/register
```
Comment (texasmichelle) [inline pipelines/azurepipeline/code/register/build.sh]: Add `#!`

File: pipelines/azurepipeline/code/training/build.sh [hunk 5]
```diff
@@ -0,0 +1,2 @@
+IMAGE=tacoregistry.azurecr.io/kubeflow/training
```
Comment (texasmichelle) [inline pipelines/azurepipeline/code/training/build.sh]: Add `#!` line

File: pipelines/azurepipeline/code/pipeline.yaml [hunk 6]
```diff
@@ -0,0 +1,287 @@
+apiVersion: argoproj.io/v1alpha1
```
Comment (texasmichelle) [inline pipelines/azurepipeline/code/pipeline.yaml]: If this is generated, is there a reason it's included?

**Comment by rem20806** [comment]
I made the requested changes. Not sure what you mean about changing file permissions... can you clarify @texasmichelle

**Comment by texasmichelle** [comment]
This PR includes changes to file permissions unrelated to your example (e.g.  code_search/docker/ks/launch_search_index_creator_job.sh). Unless there's a compelling reason, I'm requesting that you revert those changes.

**Comment by texasmichelle** [comment]
@mohona17 @sethjuarez Please respond to this PR with `@googlebot I consent.`

**Comment by texasmichelle** [comment]
There are lint issues with the python files. Once you resolve those, tests should pass.

**Comment by mohona17** [comment]
@googlebot I consent.

**Comment by sethjuarez** [comment]
@googlebot I consent.

**Comment by sethjuarez** [comment]
@googlebot I consent.

**Comment by texasmichelle** [comment]
Even though the check is red, we're good to go on the CLA front.

This PR just needs to pass presubmit tests and then it's ready to merge! Looks like lint issues are the only thing outstanding.

**Comment by sarahmaddox** [comment]
Hallo folks, can we move forward with this PR and the related docs PR? It'd be good to get them merged so that the community can use them and keep them up to date where necessary.

**Comment by rem20806** [comment]
@texasmichelle all of the lint tests have passed. Not sure what I need to do to get the tide tests to pass.

**Comment by texasmichelle** [comment]
Great work, thank you all!!

**Pull request merged**